### PR TITLE
Fixes #21111: Remove code for handling python2 in server packages

### PR DESCRIPTION
**Pull request by peckpeck**
https://issues.rudder.io/issues/21111

### CODE REVIEW

File: rudder-api-client/SPECS/dependencies [hunk 1]
```diff
@@ -1,4 +1,2 @@
-sles-12:python python-requests
 sles-15:python3 python3-requests python3-docopt python3-urllib3
-rhel-7:python python-requests python-urllib3
```
Comment (amousset) [inline rudder-api-client/SPECS/dependencies:3]: dont't we need to add the python3 version to hrhel7 too?

File: rudder-relay/SPECS/rudder-relay.spec [hunk 2]
```diff
@@ -70,62 +70,25 @@ Requires: %(../format-dependencies rpm %{old_epoch}:%{real_version} rudder-agent
...
 run a Rudder relay server on a machine.
 
-#=================================================
-# Source preparation
-#=================================================
-%prep
-%setup -c
-
-# We don't know the exact version
-cd rudder-sources-*/rudder/relay/sources/
```
Comment (amousset) [inline rudder-relay/SPECS/rudder-relay.spec:122]: why removing this?

Comment (amousset) [reply rudder-relay/SPECS/rudder-relay.spec:122]: Indeed, but I think the setup -c is still required to correctly uncompress the source

File: rudder-api-client/SPECS/dependencies (hunk 1 above)
Comment (amousset) [reply rudder-api-client/SPECS/dependencies:3]: but don't we keep it?

File: rudder-relay/SPECS/rudder-relay.spec (hunk 2 above)
Comment (peckpeck) [reply rudder-relay/SPECS/rudder-relay.spec:122]: i thought it was the default

Comment (amousset) [reply rudder-relay/SPECS/rudder-relay.spec:122]: the -c is not

**Comment by amousset** [comment]
won't be done in 7.2, should be done for 7.3, closing for now

**Pull request closed without merging**